# Amazon_data

## Part 1: Database and Jupyter Notebook Set Up
### Using following command to import dataset into mongo_db:

 mongoimport --type csv -d Clean_Data_Resources -c amazon_cleanData --headerline --drop amazon_cleanData.csv

In [51]:
# Import dependencies
from pymongo import MongoClient
from pprint import pprint
import pandas as pd

In [52]:
import matplotlib.pyplot as plt
import pandas as pd
import plotly.express as px


In [53]:
# app.py
# Import necessary libraries
from dash import Dash, html, dcc
import plotly.express as px
import pandas as pd

In [54]:
import dash
import dash_core_components as dcc
import dash_html_components as html
from dash.dependencies import Input, Output

In [55]:
# Create an instance of MongoClient
mongo = MongoClient(port=27017)

In [56]:
# confirm that our new database was created
print(mongo.list_database_names())

['Clean_Data_Resources', 'Data', 'admin', 'autosaurus', 'classDB', 'config', 'epa', 'fruits_db', 'gardenDB', 'local', 'met', 'petsitly_marketing', 'uk_food']


In [57]:
# assign the uk_food database to a variable name
db = mongo['Clean_Data_Resources']

In [58]:
# review the collections in our amazon database
print(db.list_collection_names())

['amazon_cleanData']


In [59]:
# review a document in the establishments collection
#Display the first document in the results using pprint before filter
pprint(db.amazon_cleanData.find_one())

{'_id': ObjectId('65bbe25e1c3678ff210ffaca'),
 'about_product': '【 Fast Charger& Data Sync】-With built-in safety '
                  'proctections and four-core copper wires promote maximum '
                  'signal quality and strength and enhance charging & data '
                  'transfer speed with up to 480 mb/s transferring speed.|【 '
                  'Compatibility】-Compatible with iPhone 13, 12,11, X, 8, 7, '
                  '6, 5, iPad Air, Pro, Mini & iOS devices.|【 Sturdy & '
                  'Durable】-The jacket and enforced connector made of TPE and '
                  'premium copper, are resistant to repeatedly bending and '
                  'coiling.|【 Ultra High Quality】: According to the '
                  'experimental results, the fishbone design can accept at '
                  'least 20,000 bending and insertion tests for extra '
                  'protection and durability. Upgraded 3D aluminum connector '
                  'and exclusive laser welding

In [60]:
# pprint(db.amazon_cleanData.findall)

In [61]:
# assign the collection to a variable
amazon = db['amazon_cleanData']

In [62]:
# Display the total number of items in the amazon collection
amazon.count_documents({})

1462

In [63]:
#  Create a query that finds all computer USB cables in the 'amazon' collection
# Filter results by name
query = {'category': 'ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCablesUSBCables'}

# Capture the results to a variable
results1 = amazon.find(query)

# Print the number of results
print("Total Number of items which are computer USB cables in result are:", amazon.count_documents(query))


Total Number of items which are computer USB cables in result are: 231


In [64]:
## Pretty print the results
pprint(results1[0:10])

#displaying results1
for i in results1:
    print(i)

{'_id': ObjectId('65bbe25e1c3678ff210ffaca'), 'product_id': 'B096MSW6CT', 'product_name': 'Sounce Fast Phone Charging Cable & Data Sync USB Cable Compatible for iPhone 13, 12,11, X, 8, 7, 6, 5, iPad Air, Pro, Mini & iOS Devices', 'category': 'ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCablesUSBCables', 'discounted_price': 199, 'actual_price': 1899, 'discount_percentage': 0.9, 'rating': 3, 'rating_count': 7928, 'about_product': "【 Fast Charger& Data Sync】-With built-in safety proctections and four-core copper wires promote maximum signal quality and strength and enhance charging & data transfer speed with up to 480 mb/s transferring speed.|【 Compatibility】-Compatible with iPhone 13, 12,11, X, 8, 7, 6, 5, iPad Air, Pro, Mini & iOS devices.|【 Sturdy & Durable】-The jacket and enforced connector made of TPE and premium copper, are resistant to repeatedly bending and coiling.|【 Ultra High Quality】: According to the experimental results, the fishbone design can accept at least

In [65]:
#   Create a query that finds top 10 computer USB cables in the 'amazon' collection which has discounted price greater than equal to 100 rupees and have rating score greater than equal to 4
# Filter results by name
query = {'category': 'ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCablesUSBCables',
        'discounted_price': {'$gte': 100}, 'rating':{'$gte':4}}

# Display the  fields
fields = {'category' :1, 'product_id':1, 'product_name': 1 , 'discounted_price': 1, 'discount_percentage':1,'actual_price': 1, "rating":1,"review_title":1, "review_content":1, "rating_count":1}

# sort in descending order by 
sort = [('discounted_price', -1),('actual_price', -1), ('rating', -1)]

# limit the results to the first 10
limit = 10

# Pretty print the results
pprint(list(amazon.find(query, fields).sort(sort).limit(limit)))


[{'_id': ObjectId('65bbe25e1c3678ff210ffb42'),
  'actual_price': 1999,
  'category': 'ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCablesUSBCables',
  'discount_percentage': 0.2,
  'discounted_price': 1599,
  'product_id': 'B084MZXJNK',
  'product_name': 'Belkin Apple Certified Lightning To Type C Cable, Tough '
                  'Unbreakable Braided Fast Charging For Iphone, Ipad, Air '
                  'Pods, 3.3 Feet (1 Meters)    White',
  'rating': 4,
  'rating_count': 1951,
  'review_content': 'Good cable. got original product and it is durable and '
                    'light weight,Quality is top notch and also tough as hell '
                    'very much durable and charges the devices just like the '
                    'original brand cable. Never have to worry about wear and '
                    'tear of cables.,Fast charging, built quality super,Using '
                    'to charge I phone X with Samsung 25 watt charger, it '
                    'works 

In [66]:
#displaying the result
result2 = list(amazon.find(query, fields).sort(sort))
result2

[{'_id': ObjectId('65bbe25e1c3678ff210ffb42'),
  'product_id': 'B084MZXJNK',
  'product_name': 'Belkin Apple Certified Lightning To Type C Cable, Tough Unbreakable Braided Fast Charging For Iphone, Ipad, Air Pods, 3.3 Feet (1 Meters)    White',
  'category': 'ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCablesUSBCables',
  'discounted_price': 1599,
  'actual_price': 1999,
  'discount_percentage': 0.2,
  'rating': 4,
  'rating_count': 1951,
  'review_title': 'Perfect product,Better than the original cable,Better then original,Good,The Very Best,Works absolutely fine,Charging is very slow.,Best Product',
  'review_content': 'Good cable. got original product and it is durable and light weight,Quality is top notch and also tough as hell very much durable and charges the devices just like the original brand cable. Never have to worry about wear and tear of cables.,Fast charging, built quality super,Using to charge I phone X with Samsung 25 watt charger, it works fine, charges 

In [67]:
# Initialize an empty list to store the top 10 computer USB Cables which has discounted price greater than equal to 100 and have ratings greater than equal to 4
ten_comp_USB_OVER_100 = []

# Loop through each document in the results and append it to the list
for item in result2:
   ten_comp_USB_OVER_100.append(item)
  
# Now, converted_results is a list of dictionaries
pprint(ten_comp_USB_OVER_100)

[{'_id': ObjectId('65bbe25e1c3678ff210ffb42'),
  'actual_price': 1999,
  'category': 'ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCablesUSBCables',
  'discount_percentage': 0.2,
  'discounted_price': 1599,
  'product_id': 'B084MZXJNK',
  'product_name': 'Belkin Apple Certified Lightning To Type C Cable, Tough '
                  'Unbreakable Braided Fast Charging For Iphone, Ipad, Air '
                  'Pods, 3.3 Feet (1 Meters)    White',
  'rating': 4,
  'rating_count': 1951,
  'review_content': 'Good cable. got original product and it is durable and '
                    'light weight,Quality is top notch and also tough as hell '
                    'very much durable and charges the devices just like the '
                    'original brand cable. Never have to worry about wear and '
                    'tear of cables.,Fast charging, built quality super,Using '
                    'to charge I phone X with Samsung 25 watt charger, it '
                    'works 

In [68]:
# Create a DataFrame from the list 
df_Top_comp_usb_over_100 = pd.DataFrame(ten_comp_USB_OVER_100)

# Display the DataFrame
print(df_Top_comp_usb_over_100)

                          _id  product_id  \
0    65bbe25e1c3678ff210ffb42  B084MZXJNK   
1    65bbe25e1c3678ff210ffc13  B086JTMRYL   
2    65bbe25e1c3678ff210ffb66  B084N133Y7   
3    65bbe25e1c3678ff210ffbd0  B084N1BM9L   
4    65bbe25e1c3678ff210ffb5c  B082T6GXS5   
..                        ...         ...   
186  65bbe25e1c3678ff210ffc02  B09SB6SJB4   
187  65bbe25e1c3678ff210ffb8e  B08P9RYPLR   
188  65bbe25e1c3678ff210ffb13  B08R69VDHT   
189  65bbe25e1c3678ff210ffcfc  B08R69VDHT   
190  65bbe25e1c3678ff210ffe9f  B08R69VDHT   

                                          product_name  \
0    Belkin Apple Certified Lightning To Type C Cab...   
1    ESR USB C to Lightning Cable, 10 ft (3 m), MFi...   
2    Belkin Apple Certified Lightning To Type C Cab...   
3    Belkin Apple Certified Lightning to USB Charge...   
4    AmazonBasics New Release Nylon USB-A to Lightn...   
..                                                 ...   
186  Amazon Brand - Solimo Fast Charging Braided Ty..

In [69]:
#displaying dataframe containing top 10 computer USB Cables which has discounted price greater than equal to 100 and have ratings greater than equal to 4
df_Top_comp_usb_over_100.head(10)

_id  product_id  \
0  65bbe25e1c3678ff210ffb42  B084MZXJNK   
1  65bbe25e1c3678ff210ffc13  B086JTMRYL   
2  65bbe25e1c3678ff210ffb66  B084N133Y7   
3  65bbe25e1c3678ff210ffbd0  B084N1BM9L   
4  65bbe25e1c3678ff210ffb5c  B082T6GXS5   
5  65bbe25e1c3678ff210ffb90  B084MZXJN6   
6  65bbe25e1c3678ff210ffae6  B09W5XR9RT   
7  65bbe25e1c3678ff210ffaf0  B09C6HWG18   
8  65bbe25e1c3678ff210ffcc0  B09W5XR9RT   
9  65bbe25e1c3678ff210ffdd8  B09W5XR9RT   

                                        product_name  \
0  Belkin Apple Certified Lightning To Type C Cab...   
1  ESR USB C to Lightning Cable, 10 ft (3 m), MFi...   
2  Belkin Apple Certified Lightning To Type C Cab...   
3  Belkin Apple Certified Lightning to USB Charge...   
4  AmazonBasics New Release Nylon USB-A to Lightn...   
5  Belkin Apple Certified Lightning to USB Charge...   
6  Duracell USB C To Lightning Apple Certified (M...   
7  Duracell Type C To Type C 5A (100W) Braided Sy...   
8  Duracell USB C To Lightning Apple Certified (M...   
9  Duracell USB C To Lightning Apple Certified (M...   

                                            category  discounted_price  \
0  ComputersAccessoriesAccessoriesPeripheralsCabl...              1599   
1  ComputersAccessoriesAccessoriesPeripheralsCabl...              1519   
2  ComputersAccessoriesAccessoriesPeripheralsCabl...              1499   
3  ComputersAccessoriesAccessoriesPeripheralsCabl...              1299   
4  ComputersAccessoriesAccessoriesPeripheralsCabl...               999   
5  ComputersAccessoriesAccessoriesPeripheralsCabl...               999   
6  ComputersAccessoriesAccessoriesPeripheralsCabl...               970   
7  ComputersAccessoriesAccessoriesPeripheralsCabl...               970   
8  ComputersAccessoriesAccessoriesPeripheralsCabl...               970   
9  ComputersAccessoriesAccessoriesPeripheralsCabl...               970   

   actual_price  discount_percentage  rating  rating_count  \
0          1999                 0.20       4          1951   
1          1899                 0.20       4         19763   
2          1999                 0.25       4          1951   
3          1999                 0.35       4          7318   
4          2100                 0.52       4          5492   
5          1699                 0.41       4          7318   
6          1999                 0.51       4           184   
7          1999                 0.51       4           462   
8          1999                 0.51       4           184   
9          1999                 0.51       4           184   

                                        review_title  \
0  Perfect product,Better than the original cable...   
1  Go for it,Fast charging,Good product,Good,So F...   
2  Perfect product,Better than the original cable...   
3  You might be able to get away by using other u...   
4  Good,QUALITY IS GOOD,Value for money product,V...   
5  You might be able to get away by using other u...   
6  Very good product.,Using as a spare cable in c...   
7  Product is as expected,Cable has problem with ...   
8  Very good product.,Using as a spare cable in c...   
9  Very good product.,Using as a spare cable in c...   

                                      review_content  
0  Good cable. got original product and it is dur...  
1  Everything is great go for it after all it’s e...  
2  Good cable. got original product and it is dur...  
3  Recently ordered other. Usb for just 150 rsThe...  
4  It works more than expected , little bit lose ...  
5  Recently ordered other. Usb for just 150 rsThe...  
6  Fast charging.,Cable seems to be of good quali...  
7  Same type is available in half prize other mak...  
8  Fast charging.,Cable seems to be of good quali...  
9  Fast charging.,Cable seems to be of good quali...

In [70]:
#checking column names
print(df_Top_comp_usb_over_100.columns)

Index(['_id', 'product_id', 'product_name', 'category', 'discounted_price',
       'actual_price', 'discount_percentage', 'rating', 'rating_count',
       'review_title', 'review_content'],
      dtype='object')


In [71]:
# Reorder the columns
Reordered_Top_comp_usb_over_100 = df_Top_comp_usb_over_100[[
    "product_id", "product_name", "category", "actual_price", "discounted_price",
    "discount_percentage", "rating", "review_title", "review_content","rating_count"
]]

# Print the first 10 rows of the DataFrame
print("The Top  10 computer USB Cables which has discounted price greater than equal to 100 and have ratings greater than equal to 4 are following:")
Reordered_Top_comp_usb_over_100.head(10)

The Top  10 computer USB Cables which has discounted price greater than equal to 100 and have ratings greater than equal to 4 are following:


product_id                                       product_name  \
0  B084MZXJNK  Belkin Apple Certified Lightning To Type C Cab...   
1  B086JTMRYL  ESR USB C to Lightning Cable, 10 ft (3 m), MFi...   
2  B084N133Y7  Belkin Apple Certified Lightning To Type C Cab...   
3  B084N1BM9L  Belkin Apple Certified Lightning to USB Charge...   
4  B082T6GXS5  AmazonBasics New Release Nylon USB-A to Lightn...   
5  B084MZXJN6  Belkin Apple Certified Lightning to USB Charge...   
6  B09W5XR9RT  Duracell USB C To Lightning Apple Certified (M...   
7  B09C6HWG18  Duracell Type C To Type C 5A (100W) Braided Sy...   
8  B09W5XR9RT  Duracell USB C To Lightning Apple Certified (M...   
9  B09W5XR9RT  Duracell USB C To Lightning Apple Certified (M...   

                                            category  actual_price  \
0  ComputersAccessoriesAccessoriesPeripheralsCabl...          1999   
1  ComputersAccessoriesAccessoriesPeripheralsCabl...          1899   
2  ComputersAccessoriesAccessoriesPeripheralsCabl...          1999   
3  ComputersAccessoriesAccessoriesPeripheralsCabl...          1999   
4  ComputersAccessoriesAccessoriesPeripheralsCabl...          2100   
5  ComputersAccessoriesAccessoriesPeripheralsCabl...          1699   
6  ComputersAccessoriesAccessoriesPeripheralsCabl...          1999   
7  ComputersAccessoriesAccessoriesPeripheralsCabl...          1999   
8  ComputersAccessoriesAccessoriesPeripheralsCabl...          1999   
9  ComputersAccessoriesAccessoriesPeripheralsCabl...          1999   

   discounted_price  discount_percentage  rating  \
0              1599                 0.20       4   
1              1519                 0.20       4   
2              1499                 0.25       4   
3              1299                 0.35       4   
4               999                 0.52       4   
5               999                 0.41       4   
6               970                 0.51       4   
7               970                 0.51       4   
8               970                 0.51       4   
9               970                 0.51       4   

                                        review_title  \
0  Perfect product,Better than the original cable...   
1  Go for it,Fast charging,Good product,Good,So F...   
2  Perfect product,Better than the original cable...   
3  You might be able to get away by using other u...   
4  Good,QUALITY IS GOOD,Value for money product,V...   
5  You might be able to get away by using other u...   
6  Very good product.,Using as a spare cable in c...   
7  Product is as expected,Cable has problem with ...   
8  Very good product.,Using as a spare cable in c...   
9  Very good product.,Using as a spare cable in c...   

                                      review_content  rating_count  
0  Good cable. got original product and it is dur...          1951  
1  Everything is great go for it after all it’s e...         19763  
2  Good cable. got original product and it is dur...          1951  
3  Recently ordered other. Usb for just 150 rsThe...          7318  
4  It works more than expected , little bit lose ...          5492  
5  Recently ordered other. Usb for just 150 rsThe...          7318  
6  Fast charging.,Cable seems to be of good quali...           184  
7  Same type is available in half prize other mak...           462  
8  Fast charging.,Cable seems to be of good quali...           184  
9  Fast charging.,Cable seems to be of good quali...           184

In [72]:
#  Create a query that finds all TV HDMI cables in the 'amazon' collection
# Filter results by name
query = {'category': 'ElectronicsHomeTheater,TVVideoAccessoriesCablesHDMICables'}

# Capture the results to a variable
results3 = amazon.find(query)

# Print the number of results
print("Total Number of items which are TV HDMI cables in result are:", amazon.count_documents(query))


Total Number of items which are TV HDMI cables in result are: 24


In [73]:
## Pretty print the 10 records
pprint(results3[0:10])

#displaying results1
for i in results3:
    print(i)

{'_id': ObjectId('65bbe25e1c3678ff210ffad6'), 'product_id': 'B07KSMBL2H', 'product_name': 'AmazonBasics Flexible Premium HDMI Cable (Black, 4K@60Hz, 18Gbps), 3-Foot', 'category': 'ElectronicsHomeTheater,TVVideoAccessoriesCablesHDMICables', 'discounted_price': 219, 'actual_price': 700, 'discount_percentage': 0.69, 'rating': 4, 'rating_count': 426973, 'about_product': 'Flexible, lightweight HDMI cable for connecting media devices to playback display such as HDTVs, projectors, and more|Compatible with Blu-Ray players, computers, Apple TV, Roku, cable, PS4, Xbox One, and other HDMI-compatible devices|Solid copper conductors and full metal jacket shielding for durability and high-performance connectivity|Supports Ethernet, 3D, 4K video and Audio Return Channel (ARC) and meets the latest HDMI standards|Supports bandwidth up to 18Gbps and is backwards compatible with earlier versions|Includes a 3 foot cable', 'user_id': 'AEYJ5I6JZZPOJB6MGWRQOHRQLPSQ,AFY5TVFOMVHGBPBTIJODYDQRZM5Q,AE3O6366WGEQAA

In [74]:
#   Create a query that finds top 10  TV HDMI cables in the 'amazon' collection having discounted price greater than equal to 100 rupees and have rating score greater than equal to 4
# Filter results by name
query = {'category': 'ElectronicsHomeTheater,TVVideoAccessoriesCablesHDMICables',
        'discounted_price': {'$gte': 100}, 'rating':{'$gte':4}}

# Display the 'category', 'actual_price' and 'discounted_price' fields
fields = {'category' :1, 'product_id':1, 'product_name': 1 , 'discounted_price': 1, 'discount_percentage':1,'actual_price': 1, "rating":1,"review_title":1, "review_content":1, "rating_count":1}

# sort in descending order by 
sort = [('discounted_price', -1),('actual_price', -1), ('rating', -1)]

# limit the results to the first 10
limit = 10

# Pretty print the results
pprint(list(amazon.find(query, fields).sort(sort).limit(limit)))

[{'_id': ObjectId('65bbe25e1c3678ff210ffb53'),
  'actual_price': 2399,
  'category': 'ElectronicsHomeTheater,TVVideoAccessoriesCablesHDMICables',
  'discount_percentage': 0.58,
  'discounted_price': 999,
  'product_id': 'B08TGG316Z',
  'product_name': '10k 8k 4k HDMI Cable, Certified 48Gbps 1ms Ultra High Speed '
                  'HDMI 2.1 Cable 4k 120Hz 144Hz 2k 165Hz 8k 60Hz Dynamic HDR '
                  'ARC eARC DTS:X Compatible for Mac Gaming PC Soundbar TV '
                  'Monitor Laptop PS5 4 Xbox',
  'rating': 4,
  'rating_count': 3664,
  'review_content': 'Absolutely value for money. I connect it with my Samsung '
                    '55 inch UHD TV, JBL 3.1 soundbar and Jio set top box. All '
                    'is working zakkaass !! Picture quality improved compared '
                    'to my earlier cable. Now getting Dolby 5.1 with this set '
                    'up. Thumbs up for this item.,Best product at fair price. '
                    'Quality of the cable

In [75]:
#displaying the result
result4 = list(amazon.find(query, fields).sort(sort))
result4

[{'_id': ObjectId('65bbe25e1c3678ff210ffb53'),
  'product_id': 'B08TGG316Z',
  'product_name': '10k 8k 4k HDMI Cable, Certified 48Gbps 1ms Ultra High Speed HDMI 2.1 Cable 4k 120Hz 144Hz 2k 165Hz 8k 60Hz Dynamic HDR ARC eARC DTS:X Compatible for Mac Gaming PC Soundbar TV Monitor Laptop PS5 4 Xbox',
  'category': 'ElectronicsHomeTheater,TVVideoAccessoriesCablesHDMICables',
  'discounted_price': 999,
  'actual_price': 2399,
  'discount_percentage': 0.58,
  'rating': 4,
  'rating_count': 3664,
  'review_title': 'Fantastic Ultra High Speed HDMI cable,Amazing product,Regarding cable',
  'review_content': 'Absolutely value for money. I connect it with my Samsung 55 inch UHD TV, JBL 3.1 soundbar and Jio set top box. All is working zakkaass !! Picture quality improved compared to my earlier cable. Now getting Dolby 5.1 with this set up. Thumbs up for this item.,Best product at fair price. Quality of the cable is too good.,Must buy picture clearity upto the mark'},
 {'_id': ObjectId('65bbe25e1c3

In [76]:
# Initialize an empty list to store the top 10 TV HDMI Cables which has discounted price greater than equal to 100 and have ratings greater than equal to 4
top_tv_HDMI_OVER_100 = []

# Loop through each document in the results and append it to the list
for item in result4:
   top_tv_HDMI_OVER_100.append(item)
  
# Display the list
pprint(top_tv_HDMI_OVER_100)

[{'_id': ObjectId('65bbe25e1c3678ff210ffb53'),
  'actual_price': 2399,
  'category': 'ElectronicsHomeTheater,TVVideoAccessoriesCablesHDMICables',
  'discount_percentage': 0.58,
  'discounted_price': 999,
  'product_id': 'B08TGG316Z',
  'product_name': '10k 8k 4k HDMI Cable, Certified 48Gbps 1ms Ultra High Speed '
                  'HDMI 2.1 Cable 4k 120Hz 144Hz 2k 165Hz 8k 60Hz Dynamic HDR '
                  'ARC eARC DTS:X Compatible for Mac Gaming PC Soundbar TV '
                  'Monitor Laptop PS5 4 Xbox',
  'rating': 4,
  'rating_count': 3664,
  'review_content': 'Absolutely value for money. I connect it with my Samsung '
                    '55 inch UHD TV, JBL 3.1 soundbar and Jio set top box. All '
                    'is working zakkaass !! Picture quality improved compared '
                    'to my earlier cable. Now getting Dolby 5.1 with this set '
                    'up. Thumbs up for this item.,Best product at fair price. '
                    'Quality of the cable

In [77]:
# Create a DataFrame from the list 
df_Top_TV_hdmi_over_100 = pd.DataFrame(top_tv_HDMI_OVER_100)

# Display the DataFrame
print(df_Top_TV_hdmi_over_100)

                         _id  product_id  \
0   65bbe25e1c3678ff210ffb53  B08TGG316Z   
1   65bbe25e1c3678ff210ffb47  B015OW3M1W   
2   65bbe25e1c3678ff210ffc00  B09HCH3JZG   
3   65bbe25e1c3678ff210ffb68  B07VSG5SXZ   
4   65bbe25e1c3678ff210ffbec  B07RY2X9MP   
5   65bbe25e1c3678ff210ffbee  B0BBMGLQDW   
6   65bbe25e1c3678ff210ffbad  B0718ZN31Q   
7   65bbe25e1c3678ff210ffbfa  B07RX14W1Q   
8   65bbe25e1c3678ff210ffb96  B00GG59HU2   
9   65bbe25e1c3678ff210ffbf6  B01M5967SY   
10  65bbe25e1c3678ff210ffb0b  B014I8SX4Y   
11  65bbe25e1c3678ff210ffaf9  B014I8SSD0   
12  65bbe25e1c3678ff210ffbce  B01D5H90L4   
13  65bbe25e1c3678ff210ffc09  B07VJ9ZTXS   
14  65bbe25e1c3678ff210ffbd5  B075ZTJ9XR   
15  65bbe25e1c3678ff210ffb48  B01D5H8ZI8   
16  65bbe25e1c3678ff210ffad6  B07KSMBL2H   
17  65bbe25e1c3678ff210ffd74  B07KSMBL2H   
18  65bbe25e1c3678ff210ffae3  B01M4GGIVU   
19  65bbe25e1c3678ff210ffda3  B01M4GGIVU   
20  65bbe25e1c3678ff210ffbf7  B016MDK4F4   
21  65bbe25e1c3678ff210ffba3  B0

In [78]:
df_Top_TV_hdmi_over_100.head(10)

_id  product_id  \
0  65bbe25e1c3678ff210ffb53  B08TGG316Z   
1  65bbe25e1c3678ff210ffb47  B015OW3M1W   
2  65bbe25e1c3678ff210ffc00  B09HCH3JZG   
3  65bbe25e1c3678ff210ffb68  B07VSG5SXZ   
4  65bbe25e1c3678ff210ffbec  B07RY2X9MP   
5  65bbe25e1c3678ff210ffbee  B0BBMGLQDW   
6  65bbe25e1c3678ff210ffbad  B0718ZN31Q   
7  65bbe25e1c3678ff210ffbfa  B07RX14W1Q   
8  65bbe25e1c3678ff210ffb96  B00GG59HU2   
9  65bbe25e1c3678ff210ffbf6  B01M5967SY   

                                        product_name  \
0  10k 8k 4k HDMI Cable, Certified 48Gbps 1ms Ult...   
1  AmazonBasics 6-Feet DisplayPort (not USB port)...   
2  Bestor ® 8K Hdmi 2.1 Cable 48Gbps 9.80Ft/Ultra...   
3  ZEBRONICS HAA2021 HDMI version 2.1 cable with ...   
4  AmazonBasics 10.2 Gbps High-Speed 4K HDMI Cabl...   
5  Tuarso 8K HDMI 2.1 Cable 48Gbps , 1.5 Meter Hi...   
6  Rts™ High Speed 3D Full HD 1080p Support (10 M...   
7  Amazon Basics 10.2 Gbps High-Speed 4K HDMI Cab...   
8  BlueRigger High Speed HDMI Cable with Ethernet...   
9  TIZUM High Speed HDMI Cable Aura -Gold Plated-...   

                                            category  discounted_price  \
0  ElectronicsHomeTheater,TVVideoAccessoriesCable...               999   
1  ElectronicsHomeTheater,TVVideoAccessoriesCable...               799   
2  ElectronicsHomeTheater,TVVideoAccessoriesCable...               699   
3  ElectronicsHomeTheater,TVVideoAccessoriesCable...               637   
4  ElectronicsHomeTheater,TVVideoAccessoriesCable...               609   
5  ElectronicsHomeTheater,TVVideoAccessoriesCable...               599   
6  ElectronicsHomeTheater,TVVideoAccessoriesCable...               598   
7  ElectronicsHomeTheater,TVVideoAccessoriesCable...               499   
8  ElectronicsHomeTheater,TVVideoAccessoriesCable...               467   
9  ElectronicsHomeTheater,TVVideoAccessoriesCable...               379   

   actual_price  discount_percentage  rating  rating_count  \
0          2399                 0.58       4          3664   
1          1700                 0.53       4         28638   
2          1899                 0.63       4           390   
3          1499                 0.58       4            24   
4          1500                 0.59       4          1029   
5          1999                 0.70       4            47   
6          4999                 0.88       4           910   
7           900                 0.45       4          2165   
8           599                 0.22       4         44054   
9           999                 0.62       4         12153   

                                        review_title  \
0  Fantastic Ultra High Speed HDMI cable,Amazing ...   
1  Worth Buying,Good one!!,Very nice seller servi...   
2  Great value for ultra hi speed HDMI!,thickness...   
3  good,Worth product,Good quality,Very good HDMI...   
4  AmazonBasics Product,Good,Very good,Value for ...   
5  Good Product,Nice,Customer service support inf...   
6  Good quality but not superb,Good,Good,Not work...   
7  Sturdy,Super,Working good,Always go for qualit...   
8  BEST WITH BOAT & LG SMART TV,This product is o...   
9  Cheap product and same is the performance but ...   

                                      review_content  
0  Absolutely value for money. I connect it with ...  
1  It's really worth it . I am using it to get ou...  
2  Would have liked the product to be in the comp...  
3  good,Value for money 👍,This is really a good q...  
4  The product quality is really amazing. The wir...  
5  As mention in description, its awesome.,Nice,G...  
6  Like ok,Product as specified,Good,,Yes it's ve...  
7  Works well. Sturdy built and good quality. Onl...  
8  Perfect hdmi cable for boat soundbar and lg sm...  
9  The signal is too unstable once connect you ne...

In [79]:
# Reorder the columns
Reordered_Top_tv_hdmi_over_100 = df_Top_TV_hdmi_over_100[[
    "product_id", "product_name", "category", "actual_price", "discounted_price",
    "discount_percentage", "rating", "review_title", "review_content","rating_count"
]]

# Print the first 10 rows of the DataFrame
print("The Top  10 TV HDMI Cables which has discounted price greater than equal to 100 and have ratings greater than equal to 4 are following:")
Reordered_Top_tv_hdmi_over_100.head(10)

The Top  10 TV HDMI Cables which has discounted price greater than equal to 100 and have ratings greater than equal to 4 are following:


product_id                                       product_name  \
0  B08TGG316Z  10k 8k 4k HDMI Cable, Certified 48Gbps 1ms Ult...   
1  B015OW3M1W  AmazonBasics 6-Feet DisplayPort (not USB port)...   
2  B09HCH3JZG  Bestor ® 8K Hdmi 2.1 Cable 48Gbps 9.80Ft/Ultra...   
3  B07VSG5SXZ  ZEBRONICS HAA2021 HDMI version 2.1 cable with ...   
4  B07RY2X9MP  AmazonBasics 10.2 Gbps High-Speed 4K HDMI Cabl...   
5  B0BBMGLQDW  Tuarso 8K HDMI 2.1 Cable 48Gbps , 1.5 Meter Hi...   
6  B0718ZN31Q  Rts™ High Speed 3D Full HD 1080p Support (10 M...   
7  B07RX14W1Q  Amazon Basics 10.2 Gbps High-Speed 4K HDMI Cab...   
8  B00GG59HU2  BlueRigger High Speed HDMI Cable with Ethernet...   
9  B01M5967SY  TIZUM High Speed HDMI Cable Aura -Gold Plated-...   

                                            category  actual_price  \
0  ElectronicsHomeTheater,TVVideoAccessoriesCable...          2399   
1  ElectronicsHomeTheater,TVVideoAccessoriesCable...          1700   
2  ElectronicsHomeTheater,TVVideoAccessoriesCable...          1899   
3  ElectronicsHomeTheater,TVVideoAccessoriesCable...          1499   
4  ElectronicsHomeTheater,TVVideoAccessoriesCable...          1500   
5  ElectronicsHomeTheater,TVVideoAccessoriesCable...          1999   
6  ElectronicsHomeTheater,TVVideoAccessoriesCable...          4999   
7  ElectronicsHomeTheater,TVVideoAccessoriesCable...           900   
8  ElectronicsHomeTheater,TVVideoAccessoriesCable...           599   
9  ElectronicsHomeTheater,TVVideoAccessoriesCable...           999   

   discounted_price  discount_percentage  rating  \
0               999                 0.58       4   
1               799                 0.53       4   
2               699                 0.63       4   
3               637                 0.58       4   
4               609                 0.59       4   
5               599                 0.70       4   
6               598                 0.88       4   
7               499                 0.45       4   
8               467                 0.22       4   
9               379                 0.62       4   

                                        review_title  \
0  Fantastic Ultra High Speed HDMI cable,Amazing ...   
1  Worth Buying,Good one!!,Very nice seller servi...   
2  Great value for ultra hi speed HDMI!,thickness...   
3  good,Worth product,Good quality,Very good HDMI...   
4  AmazonBasics Product,Good,Very good,Value for ...   
5  Good Product,Nice,Customer service support inf...   
6  Good quality but not superb,Good,Good,Not work...   
7  Sturdy,Super,Working good,Always go for qualit...   
8  BEST WITH BOAT & LG SMART TV,This product is o...   
9  Cheap product and same is the performance but ...   

                                      review_content  rating_count  
0  Absolutely value for money. I connect it with ...          3664  
1  It's really worth it . I am using it to get ou...         28638  
2  Would have liked the product to be in the comp...           390  
3  good,Value for money 👍,This is really a good q...            24  
4  The product quality is really amazing. The wir...          1029  
5  As mention in description, its awesome.,Nice,G...            47  
6  Like ok,Product as specified,Good,,Yes it's ve...           910  
7  Works well. Sturdy built and good quality. Onl...          2165  
8  Perfect hdmi cable for boat soundbar and lg sm...         44054  
9  The signal is too unstable once connect you ne...         12153

In [80]:
# Print the first 10 rows of the DataFrame
print("The Top  10 TV HDMI Cables which has discounted price greater than equal to 100 and have ratings greater than equal to 4 are following:")
Reordered_Top_tv_hdmi_over_100.head(10)

The Top  10 TV HDMI Cables which has discounted price greater than equal to 100 and have ratings greater than equal to 4 are following:


product_id                                       product_name  \
0  B08TGG316Z  10k 8k 4k HDMI Cable, Certified 48Gbps 1ms Ult...   
1  B015OW3M1W  AmazonBasics 6-Feet DisplayPort (not USB port)...   
2  B09HCH3JZG  Bestor ® 8K Hdmi 2.1 Cable 48Gbps 9.80Ft/Ultra...   
3  B07VSG5SXZ  ZEBRONICS HAA2021 HDMI version 2.1 cable with ...   
4  B07RY2X9MP  AmazonBasics 10.2 Gbps High-Speed 4K HDMI Cabl...   
5  B0BBMGLQDW  Tuarso 8K HDMI 2.1 Cable 48Gbps , 1.5 Meter Hi...   
6  B0718ZN31Q  Rts™ High Speed 3D Full HD 1080p Support (10 M...   
7  B07RX14W1Q  Amazon Basics 10.2 Gbps High-Speed 4K HDMI Cab...   
8  B00GG59HU2  BlueRigger High Speed HDMI Cable with Ethernet...   
9  B01M5967SY  TIZUM High Speed HDMI Cable Aura -Gold Plated-...   

                                            category  actual_price  \
0  ElectronicsHomeTheater,TVVideoAccessoriesCable...          2399   
1  ElectronicsHomeTheater,TVVideoAccessoriesCable...          1700   
2  ElectronicsHomeTheater,TVVideoAccessoriesCable...          1899   
3  ElectronicsHomeTheater,TVVideoAccessoriesCable...          1499   
4  ElectronicsHomeTheater,TVVideoAccessoriesCable...          1500   
5  ElectronicsHomeTheater,TVVideoAccessoriesCable...          1999   
6  ElectronicsHomeTheater,TVVideoAccessoriesCable...          4999   
7  ElectronicsHomeTheater,TVVideoAccessoriesCable...           900   
8  ElectronicsHomeTheater,TVVideoAccessoriesCable...           599   
9  ElectronicsHomeTheater,TVVideoAccessoriesCable...           999   

   discounted_price  discount_percentage  rating  \
0               999                 0.58       4   
1               799                 0.53       4   
2               699                 0.63       4   
3               637                 0.58       4   
4               609                 0.59       4   
5               599                 0.70       4   
6               598                 0.88       4   
7               499                 0.45       4   
8               467                 0.22       4   
9               379                 0.62       4   

                                        review_title  \
0  Fantastic Ultra High Speed HDMI cable,Amazing ...   
1  Worth Buying,Good one!!,Very nice seller servi...   
2  Great value for ultra hi speed HDMI!,thickness...   
3  good,Worth product,Good quality,Very good HDMI...   
4  AmazonBasics Product,Good,Very good,Value for ...   
5  Good Product,Nice,Customer service support inf...   
6  Good quality but not superb,Good,Good,Not work...   
7  Sturdy,Super,Working good,Always go for qualit...   
8  BEST WITH BOAT & LG SMART TV,This product is o...   
9  Cheap product and same is the performance but ...   

                                      review_content  rating_count  
0  Absolutely value for money. I connect it with ...          3664  
1  It's really worth it . I am using it to get ou...         28638  
2  Would have liked the product to be in the comp...           390  
3  good,Value for money 👍,This is really a good q...            24  
4  The product quality is really amazing. The wir...          1029  
5  As mention in description, its awesome.,Nice,G...            47  
6  Like ok,Product as specified,Good,,Yes it's ve...           910  
7  Works well. Sturdy built and good quality. Onl...          2165  
8  Perfect hdmi cable for boat soundbar and lg sm...         44054  
9  The signal is too unstable once connect you ne...         12153

In [81]:
#  Create a query that finds all TV USB cables in the 'amazon' collection
# Filter results by name

query = {'category': {'$regex': "Cables"}}

# Capture the results to a variable
results5 = amazon.find(query)

# Print the number of results
print("Total Number of cables in result are:", amazon.count_documents(query))


Total Number of cables in result are: 270


In [82]:
## Pretty print the 10 records
Cables = []
pprint(results5)

#displaying results1
for i in results5:
    Cables.append(i)
   

In [83]:
Cables[0:3]

[{'_id': ObjectId('65bbe25e1c3678ff210ffaca'),
  'product_id': 'B096MSW6CT',
  'product_name': 'Sounce Fast Phone Charging Cable & Data Sync USB Cable Compatible for iPhone 13, 12,11, X, 8, 7, 6, 5, iPad Air, Pro, Mini & iOS Devices',
  'category': 'ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCablesUSBCables',
  'discounted_price': 199,
  'actual_price': 1899,
  'discount_percentage': 0.9,
  'rating': 3,
  'rating_count': 7928,
  'about_product': "【 Fast Charger& Data Sync】-With built-in safety proctections and four-core copper wires promote maximum signal quality and strength and enhance charging & data transfer speed with up to 480 mb/s transferring speed.|【 Compatibility】-Compatible with iPhone 13, 12,11, X, 8, 7, 6, 5, iPad Air, Pro, Mini & iOS devices.|【 Sturdy & Durable】-The jacket and enforced connector made of TPE and premium copper, are resistant to repeatedly bending and coiling.|【 Ultra High Quality】: According to the experimental results, the fishbone design 

In [84]:
# Create a DataFrame from the list 
df_cables = pd.DataFrame(Cables)

#reorder the df_Cables dataframe
reordered_df_cables = df_cables[["category", "product_name", "discounted_price", "actual_price", "rating", "rating_count"]]

# Display the DataFrame
reordered_df_cables

category  \
0    ComputersAccessoriesAccessoriesPeripheralsCabl...   
1    ComputersAccessoriesAccessoriesPeripheralsCabl...   
2    ComputersAccessoriesAccessoriesPeripheralsCabl...   
3    ComputersAccessoriesAccessoriesPeripheralsCabl...   
4    ComputersAccessoriesAccessoriesPeripheralsCabl...   
..                                                 ...   
265  ComputersAccessoriesAccessoriesPeripheralsCabl...   
266  ComputersAccessoriesAccessoriesPeripheralsCabl...   
267  ComputersAccessoriesAccessoriesPeripheralsCabl...   
268  ComputersAccessoriesAccessoriesPeripheralsCabl...   
269  ComputersAccessoriesAccessoriesPeripheralsCabl...   

                                          product_name  discounted_price  \
0    Sounce Fast Phone Charging Cable & Data Sync U...               199   
1    Wayona Nylon Braided USB to Lightning Fast Cha...               399   
2    Ambrane Unbreakable 60W / 3A Fast Charging 1.5...               199   
3    Portronics Konnect L 1.2M Fast Charging 3A 8 P...               154   
4    boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...               329   
..                                                 ...               ...   
265  Ambrane 2 in 1 Type-C & Micro USB Cable with 6...               199   
266  Ambrane 60W / 3A Fast Charging Output Cable wi...               179   
267  SWAPKART Fast Charging Cable and Data Sync USB...               209   
268  Wayona Usb Type C Fast Charger Cable Fast Char...               325   
269  Lapster 65W compatible for OnePlus Dash Warp C...               199   

     actual_price  rating  rating_count  
0            1899       3          7928  
1            1099       4         24269  
2             349       4         43994  
3             399       4         16905  
4             699       4         94363  
..            ...     ...           ...  
265           499       4           602  
266           399       4          1423  
267           499       3           536  
268          1299       4         10576  
269           999       4           127  

[270 rows x 6 columns]

In [85]:
#  Create a query that finds all TV USB cables in the 'amazon' collection
# Filter results by name
query = {'category': {'$regex': "RCACables"}}

# Capture the results to a variable
results5 = amazon.find(query)

# Print the number of results
print("Total Number of rca cables in result are:", amazon.count_documents(query))


Total Number of rca cables in result are: 2


In [86]:
#  Create a query that finds all Smart Televisions in the 'amazon' collection
# Filter results by category name
query = {'category': 'ElectronicsHomeTheater,TVVideoTelevisionsSmartTelevisions'}

# Capture the results to a variable
results6 = amazon.find(query)

# Print the number of results
print("Total Number of items which are all Smart Televisions in result are:", amazon.count_documents(query))


Total Number of items which are all Smart Televisions in result are: 63


In [87]:
## Pretty print the 10 records
pprint(results6[0:10])

#displaying results1
for i in results6:
    print(i)


{'_id': ObjectId('65bbe25e1c3678ff210ffada'), 'product_id': 'B0B6F7LX4C', 'product_name': 'MI 80 cm (32 inches) 5A Series HD Ready Smart Android LED TV L32M7-5AIN (Black)', 'category': 'ElectronicsHomeTheater,TVVideoTelevisionsSmartTelevisions', 'discounted_price': 13999, 'actual_price': 24999, 'discount_percentage': 0.44, 'rating': 4, 'rating_count': 32840, 'about_product': 'Note : The brands, Mi and Xiaomi, are part of the same multinational conglomerate|Resolution : HD Ready (1366 x 768) Resolution | Refresh Rate : 60 Hertz | 178 Degree wide viewing angle|Connectivity: Dual Band Wi-Fi | 2 HDMI ports to connect latest gaming consoles, set top box, Blu-ray Players | 2 USB ports to connect hard drives and other USB devices | ALLM | ARC | Bluetooth 5.0 | Ethernet|Sound: 20 Watts Output | Dolby Audio, DTS Virtual: X, DTS-HD|Smart TV Features : Android TV 11 | PatchWall | IMDb Integration | Universal Search | 300+ Free Live Channels | Kids Mode with Parental lock | Smart Recommendations |

In [88]:
#  Create a query that finds all Smart Televisions in the 'amazon' collection
# Filter results by category name
query = {'category': {'$regex': "SmartTelevisions"}}

# Capture the results to a variable
results7 = amazon.find(query)

# Print the number of results
print("Total Number of items which are all Smart Televisions in result are:", amazon.count_documents(query))


Total Number of items which are all Smart Televisions in result are: 63


In [89]:
#   Create a query that finds top 10  Smart Televisions in the 'amazon' collection have rating score greater than equal to 4 and highest rating count in descending order
# Filter results by product name
query = {'category': 'ElectronicsHomeTheater,TVVideoTelevisionsSmartTelevisions',
        'rating':{'$gte':4}}

# Display the 'category', 'actual_price' and 'discounted_price' fields
fields = {'category' :1, 'product_id':1, 'product_name': 1 , 'discounted_price': 1, 'discount_percentage':1,'actual_price': 1, "rating":1,"review_title":1, "review_content":1, "rating_count":1}

# sort in descending order by 
sort = [('rating', -1), ('rating_count', -1)]

limit = 10

# Using find() with sort and limit as chained method calls
pprint(list(amazon.find(query, fields).sort(sort).limit(limit)))

# Print the number of results
print("Total Number of items which are all Smart Televisions in result are:", amazon.count_documents(query))

[{'_id': ObjectId('65bbe25e1c3678ff210ffb12'),
  'actual_price': 42999,
  'category': 'ElectronicsHomeTheater,TVVideoTelevisionsSmartTelevisions',
  'discount_percentage': 0.37,
  'discounted_price': 26999,
  'product_id': 'B09RFC46VP',
  'product_name': 'Redmi 108 cm (43 inches) 4K Ultra HD Android Smart LED TV '
                  'X43 | L43R7-7AIN (Black)',
  'rating': 4,
  'rating_count': 45238,
  'review_content': 'Worth the price. I use the local cable providers set up. '
                    'With my previous tv (Samsung) I was not facing this '
                    'problem.  The only problem I am facing is each time I '
                    'turn off the tv from the mains, I have to go to the '
                    'settings and turn the settings on to tv speaker settings '
                    'despite no other speaker or WiFi or blue tooth or cell '
                    'phone being in the proximity. As a result I have stopped '
                    'turning the tv off from the main

In [90]:
#displaying the result
result8 = list(amazon.find(query, fields).sort(sort).limit(limit))
result8


[{'_id': ObjectId('65bbe25e1c3678ff210ffb12'),
  'product_id': 'B09RFC46VP',
  'product_name': 'Redmi 108 cm (43 inches) 4K Ultra HD Android Smart LED TV X43 | L43R7-7AIN (Black)',
  'category': 'ElectronicsHomeTheater,TVVideoTelevisionsSmartTelevisions',
  'discounted_price': 26999,
  'actual_price': 42999,
  'discount_percentage': 0.37,
  'rating': 4,
  'rating_count': 45238,
  'review_title': 'Worth the price,Mi Smart Tv 32" :- 7/10 average.,Worth using since 1.5 years,expect more from mi,Worth for money.,Good product,It’s good,Go for it without thinking twice.',
  'review_content': "Worth the price. I use the local cable providers set up. With my previous tv (Samsung) I was not facing this problem.  The only problem I am facing is each time I turn off the tv from the mains, I have to go to the settings and turn the settings on to tv speaker settings despite no other speaker or WiFi or blue tooth or cell phone being in the proximity. As a result I have stopped turning the tv off fro

In [91]:
# Initialize an empty list to store the top 10 TV HDMI Cables which has discounted price greater than equal to 100 and have ratings greater than equal to 4
top10_SmartTVs = []

# Loop through each document in the results and append it to the list
for item in result8:
   top10_SmartTVs.append(item)
  
# Display the list
pprint(top10_SmartTVs)


[{'_id': ObjectId('65bbe25e1c3678ff210ffb12'),
  'actual_price': 42999,
  'category': 'ElectronicsHomeTheater,TVVideoTelevisionsSmartTelevisions',
  'discount_percentage': 0.37,
  'discounted_price': 26999,
  'product_id': 'B09RFC46VP',
  'product_name': 'Redmi 108 cm (43 inches) 4K Ultra HD Android Smart LED TV '
                  'X43 | L43R7-7AIN (Black)',
  'rating': 4,
  'rating_count': 45238,
  'review_content': 'Worth the price. I use the local cable providers set up. '
                    'With my previous tv (Samsung) I was not facing this '
                    'problem.  The only problem I am facing is each time I '
                    'turn off the tv from the mains, I have to go to the '
                    'settings and turn the settings on to tv speaker settings '
                    'despite no other speaker or WiFi or blue tooth or cell '
                    'phone being in the proximity. As a result I have stopped '
                    'turning the tv off from the main

In [92]:
# Create a DataFrame from the list 
df_Top_Smart_TV = pd.DataFrame(top10_SmartTVs)

# Display the DataFrame
#print(df_Top_Smart_TV)
df_Top_Smart_TV.head(10)


_id  product_id  \
0  65bbe25e1c3678ff210ffb12  B09RFC46VP   
1  65bbe25e1c3678ff210ffb0a  B09F9YQQ7B   
2  65bbe25e1c3678ff210ffb46  B08Y55LPBF   
3  65bbe25e1c3678ff210ffe81  B09F9YQQ7B   
4  65bbe25e1c3678ff210ffb25  B09Q5P2MT3   
5  65bbe25e1c3678ff210ffb03  B09Q5SWVBJ   
6  65bbe25e1c3678ff210ffae4  B08B42LWKN   
7  65bbe25e1c3678ff210ffbe0  B09HQSV46W   
8  65bbe25e1c3678ff210ffb8f  B0B6F8HHR6   
9  65bbe25e1c3678ff210ffb30  B0B6F98KJJ   

                                        product_name  \
0  Redmi 108 cm (43 inches) 4K Ultra HD Android S...   
1  Redmi 80 cm (32 inches) Android 11 Series HD R...   
2  Redmi 126 cm (50 inches) 4K Ultra HD Android S...   
3  Redmi 80 cm (32 inches) Android 11 Series HD R...   
4  OnePlus 108 cm (43 inches) Y Series Full HD Sm...   
5  OnePlus 80 cm (32 inches) Y Series HD Ready Sm...   
6  OnePlus 80 cm (32 inches) Y Series HD Ready LE...   
7  Mi 100 cm (40 inches) Horizon Edition Full HD ...   
8  MI 108 cm (43 inches) 5A Series Full HD Smart ...   
9  MI 100 cm (40 inches) 5A Series Full HD Smart ...   

                                            category  discounted_price  \
0  ElectronicsHomeTheater,TVVideoTelevisionsSmart...             26999   
1  ElectronicsHomeTheater,TVVideoTelevisionsSmart...             13999   
2  ElectronicsHomeTheater,TVVideoTelevisionsSmart...             32999   
3  ElectronicsHomeTheater,TVVideoTelevisionsSmart...             13999   
4  ElectronicsHomeTheater,TVVideoTelevisionsSmart...             24999   
5  ElectronicsHomeTheater,TVVideoTelevisionsSmart...             15999   
6  ElectronicsHomeTheater,TVVideoTelevisionsSmart...             14999   
7  ElectronicsHomeTheater,TVVideoTelevisionsSmart...             21999   
8  ElectronicsHomeTheater,TVVideoTelevisionsSmart...             24999   
9  ElectronicsHomeTheater,TVVideoTelevisionsSmart...             21999   

   actual_price  discount_percentage  rating  rating_count  \
0         42999                 0.37       4         45238   
1         24999                 0.44       4         45238   
2         44999                 0.27       4         45238   
3         24999                 0.44       4         45237   
4         31999                 0.22       4         34899   
5         21999                 0.27       4         34899   
6         19999                 0.25       4         34899   
7         29999                 0.27       4         32840   
8         35999                 0.31       4         32840   
9         29999                 0.27       4         32840   

                                        review_title  \
0  Worth the price,Mi Smart Tv 32" :- 7/10 averag...   
1  Worth the price,Mi Smart Tv 32" :- 7/10 averag...   
2  Worth the price,Mi Smart Tv 32" :- 7/10 averag...   
3  Worth the price,Mi Smart Tv 32" :- 7/10 averag...   
4  Worthy and most affordable - Great TV,Good pro...   
5  Worthy and most affordable - Great TV,Good pro...   
6  Worthy and most affordable - Great TV,Good pro...   
7  It is the best tv if you are getting it in 10-...   
8  It is the best tv if you are getting it in 10-...   
9  It is the best tv if you are getting it in 10-...   

                                      review_content  
0  Worth the price. I use the local cable provide...  
1  Worth the price. I use the local cable provide...  
2  Worth the price. I use the local cable provide...  
3  Worth the price. I use the local cable provide...  
4  This OnePlus TV is great if you want a afforda...  
5  This OnePlus TV is great if you want a afforda...  
6  This OnePlus TV is great if you want a afforda...  
7  Pros- xiomi 5a is best in budget-Nice picture ...  
8  Pros- xiomi 5a is best in budget-Nice picture ...  
9  Pros- xiomi 5a is best in budget-Nice picture ...

In [93]:
import matplotlib.pyplot as plt
import seaborn as sns

# Statistical Data Analysis

In [94]:
# Calculating the correlation coefficient between 'discounted_price' and  "actual_price"
corr = df_Top_Smart_TV['actual_price'].corr(df_Top_Smart_TV['discounted_price'])
corr

0.9365349800625035

# Analysis

The calculated correlation coefficient between 'actual_price' and 'discounted_price' is approximately 0.94. This value indicates a strong positive linear relationship between the actual price and the discounted price of items in the dataset. In practical terms, it means that as the actual price increases, the discounted price tends to increase as well, and vice versa. This strong correlation suggests that the two prices move in tandem to a large extent, implying that the discounting strategy might be proportionally related to the actual price of the items.






In [95]:
# Calculating the correlation coefficient between  "rating_count" and 'discounted_price'
corr = df_Top_Smart_TV['rating_count'].corr(df_Top_Smart_TV['discounted_price'])
corr

0.053776443133383325

In [96]:
# Calculating the correlation coefficient between 'discounted_price' and  "rating_count"
corr = df_Top_Smart_TV['discounted_price'].corr(df_Top_Smart_TV['rating_count'])
corr

0.053776443133383325

# Analysis


The calculated correlation coefficient between 'rating_count' and 'discounted_price' is approximately 0.054. This value indicates a very weak positive linear relationship between the number of ratings a product has received and its discounted price within the dataset. Essentially, this suggests that there is barely any linear correlation between how many times a product has been rated and its discounted price. The correlation is very close to 0, implying that changes in the discounted price of items do not significantly correspond with changes in their rating count, and vice versa.






In [97]:
# Calculating the correlation coefficient between 'discounted_price' and  "actual_price"
corr = df_Top_Smart_TV['rating_count'].corr(df_Top_Smart_TV['rating'])
corr

c:\Users\PC\AppData\Local\Programs\Python\Python311\Lib\site-packages\numpy\lib\function_base.py:2897: RuntimeWarning:

invalid value encountered in divide

c:\Users\PC\AppData\Local\Programs\Python\Python311\Lib\site-packages\numpy\lib\function_base.py:2898: RuntimeWarning:

invalid value encountered in divide



nan

In [106]:
# Filter the DataFrame for TVs with a rating of 4
df_filtered = df_Top_Smart_TV[df_Top_Smart_TV['rating'] == 4]

# Transform the DataFrame to long format
df_long = df_filtered.melt(id_vars=['rating', 'product_id','product_name', 'category'], value_vars=['actual_price', 'discounted_price'],
                           var_name='Price Type', value_name='Price')


In [107]:
df_long.head()

rating  product_id                                       product_name  \
0       4  B09RFC46VP  Redmi 108 cm (43 inches) 4K Ultra HD Android S...   
1       4  B09F9YQQ7B  Redmi 80 cm (32 inches) Android 11 Series HD R...   
2       4  B08Y55LPBF  Redmi 126 cm (50 inches) 4K Ultra HD Android S...   
3       4  B09F9YQQ7B  Redmi 80 cm (32 inches) Android 11 Series HD R...   
4       4  B09Q5P2MT3  OnePlus 108 cm (43 inches) Y Series Full HD Sm...   

                                            category    Price Type  Price  
0  ElectronicsHomeTheater,TVVideoTelevisionsSmart...  actual_price  42999  
1  ElectronicsHomeTheater,TVVideoTelevisionsSmart...  actual_price  24999  
2  ElectronicsHomeTheater,TVVideoTelevisionsSmart...  actual_price  44999  
3  ElectronicsHomeTheater,TVVideoTelevisionsSmart...  actual_price  24999  
4  ElectronicsHomeTheater,TVVideoTelevisionsSmart...  actual_price  31999

In [110]:
# Create the bar chart using the long format DataFrame
fig = px.bar(df_long, x='Price Type', y='Price', color='Price Type', barmode='group', hover_data=['category','product_id','product_name' ])

# Update layout for a clearer x-axis and set the title
fig.update_layout(
    title="Discounted Price Vs Actual Prices for Smart TVs with Rating Score 4",
    xaxis_title="Price Type",
    yaxis_title="Price (in INR)",
    xaxis_tickangle=-45,
    legend_title="Price Type",
    # Make sure the currency format is correct for your locale
    # This example assumes that you are formatting the prices in Indian Rupees
)

# Show the figure
fig.show()

In [114]:
#saving the chart into the output folder
fig.write_html('../outputs_Viz/Amazon_bar1.html')


In [99]:
# data = df_Top_Smart_TV

In [100]:
# Assuming df_Top_Smart_TV is your DataFrame and it has a 'rating' column
fig = px.bar(df_Top_Smart_TV, x='actual_price', y='discounted_price', color='Price Type', barmode='group')

# Update layout for a clearer x-axis
fig.update_layout(
    title = "Discounted_price Vs Actual_prices for Smart TVs with rating score 4 and above",
    xaxis_title="Actual Price (in INR)",
    xaxis_tickmode='array',
    xaxis_tickvals=df_Top_Smart_TV['actual_price'],
    xaxis_ticktext=[f"${int(price):,}" for price in df_Top_Smart_TV['actual_price']],
    xaxis_tickangle=-45
)

fig.show()
df_filtered = df_Top_Smart_TV[df_Top_Smart_TV['rating'] == 4]


ValueError: Value of 'color' is not the name of a column in 'data_frame'. Expected one of ['_id', 'product_id', 'product_name', 'category', 'discounted_price', 'actual_price', 'discount_percentage', 'rating', 'rating_count', 'review_title', 'review_content'] but received: Price Type

In [ ]:
import plotly.express as px

In [ ]:
# plt.figure(figsize=(12, 6))

# # Bar plot for discounted price and actual price
# plt.subplot(1, 2, 1)
# sns.barplot(x=df_Top_Smart_TV["product_name"], y=df_Top_Smart_TV["discounted_price"], color="skyblue", label="Discounted Price")
# sns.barplot(x=df_Top_Smart_TV["product_name"], y=df_Top_Smart_TV["actual_price"], color="orange", label="Actual Price")
# plt.title("Discounted Price vs Actual Price")
# plt.xticks(rotation=45, ha="right")
# plt.legend()
# plt.show()

In [ ]:
# In Python:

# plotly.plotly( [plotly.graph_objs .type(x ,y ,mode , marker = dict(color ,size ))]

In [ ]:
# plt.figure(figsize=(10, 6))
# sns.scatterplot(x=df_Top_Smart_TV["rating_count"], y=df_Top_Smart_TV["discounted_price"], hue=df_Top_Smart_TV["product_name"], s=100)
# plt.title("Rating Count vs Discounted Price")
# plt.xlabel("Rating Count")
# plt.ylabel("Discounted Price")
# #plt.legend(title="Product Name")
# plt.show()

In [ ]:
# Create a pipeline that: 
# 1. Matches smart TVs having rating score greater than equal to 4
match_query = {'$match': {'category': 'ElectronicsHomeTheater,TVVideoTelevisionsSmartTelevisions',
                'rating': {'$gte': 4 }}}


# 2. Groups the matches by rating count having most rating count at the top
group_query = {'$group': {'_id':"$rating_count", 'total': {'$sum':1}}}

# 3. Sorts the matches from highest to lowest
sort_values = {'$sort': {'total':-1}}

# Put the pipeline together
pipeline = [match_query, group_query, sort_values]

In [ ]:
# Run the pipeline through the aggregate method, cast the results as a list, and save the results to a variable
result9 = list(amazon.aggregate(pipeline))

In [ ]:
# Print the first 10 results
print("Number of  Smart TVS having rating score greater than equal to 4 in result are : ", len(results))
print("The first ten results are: ")
pprint(result9[0:10])

# Analysis

### Here's a detailed explanation of the output list:

- {'_id': 32840, 'total': 5}: There are 5 smart TVs that received 32,840 ratings. This group has the highest number of TVs, indicating that 32,840 is a common rating count among the TVs considered.
- {'_id': 7109, 'total': 4}, {'_id': 4703, 'total': 4}: Each of these groups has 4 smart TVs that received 7,109 and 4,703 ratings, respectively.
- {'_id': 45238, 'total': 3}, {'_id': 16299, 'total': 3}, {'_id': 34899, 'total': 3}: Each of these groups consists of 3 smart TVs, with each group having 45,238, 16,299, and 34,899 ratings, respectively where the first one having the highest rating counts among all.
- Other entries ('total': 2): These represent groups where exactly 2 smart TVs have the specified number of ratings (_id values like 21252, 3587, etc.).
- This output essentially ranks the commonality of certain rating_count values among smart TVs in the dataset, starting from the most common (or shared by the most products) to less common. It's a valuable insight into the distribution of customer engagement (as measured by the count of ratings) across different products in the amazon data collection.

In [ ]:
# Convert the result to a Pandas DataFrame
smartTVS_df = pd.DataFrame(result9)

# Display the number of rows in the DataFrame
print("Rows in DataFrame: ", len(smartTVS_df))

# Display the first 10 rows of the DataFrame
smartTVS_df.head(10)

In [ ]:
# Rename the columns
renamed_smartTVS_df = smartTVS_df.rename(columns={"_id": "rating_count",
                                              "total": "Total_count",
                                            })

# Print the first 10 rows of the DataFrame
print(" ")
renamed_smartTVS_df.head(10)

## Analysis


The summarized output of the DataFrame indicates a grouping of smart TVs based on their rating counts, revealing insights into customer engagement or popularity. Specifically, the top group consists of 5 smart TVs sharing a high rating count of 32,840, suggesting a significant level of customer interaction. Following this, there are groups of 4 smart TVs with rating counts of 7,109 and 4,703, respectively, indicating these are also highly rated items. Further down, groups of 3 smart TVs share rating counts of 45,238, 16,299, and 34,899, demonstrating  the highest level for the first one then moderate level of popularity for the second one but again higher level of popularity for the third one. Lastly, several groups of 2 smart TVs show, first one having relatively lower, yet notable, rating counts ranging from 1,611 to 21,252. This analysis underscores the variability in customer ratings among smart TVs, from extremely popular models with thousands of ratings to less popular ones with fewer ratings.